collective/pfg/signup/tests/test_adapter.py

```diff
@@ -23,10 +23,135 @@ def test_creation(self):
         self.assertEqual(signup_adapter.portal_type, 'SignUpAdapter')
 
+    def dummy_send( self, mfrom, mto, messageText, immediate=False ):
+        self.mfrom = mfrom
+        self.mto = mto
+        self.messageText = messageText
+        self.messageBody = '\n\n'.join(messageText.split('\n\n')[1:])
+
+    def LoadRequestForm(self, **kwargs):
+        form = self.app.REQUEST.form
+        form.clear()
+        for key in kwargs.keys():
+            form[key] = kwargs[key]
+        return self.app.REQUEST
+
+    def setUp(self):
+        self.app = self.layer['app']
+        portal = self.layer['portal']
+        form = portal['form']
+        form.checkAuthenticator = False # no csrf protection
+        self.mailhost = form.MailHost
+        self.mailhost._send = self.dummy_send
+        self.form = form
+        form.invokeFactory('FormStringField', 'fullname')
+        form.invokeFactory('FormStringField', 'username')
+        form.invokeFactory('FormStringField', 'email')
+        form.invokeFactory('FormStringField', 'department')
+        form.mailer.setRecipient_name('Mail Dummy')
+        form.mailer.setRecipient_email('mdummy@address.com')
+        form.signup.full_name_field = 'fullname'
+        form.signup.username_field = 'username'
+        form.signup.email_field = 'email'
+
+    def test_auto_register(self):
+        signup = self.form.signup
+        fields = self.form._getFieldObjects()
+        request = self.LoadRequestForm(fullname='test name', username='test1', email='test@mail.com')
+        signup.onSuccess(fields, request)
+        user = api.user.get(username='test1')
+        self.assertIsNotNone(user, 'Fail to auto-register user')
+
+    def test_add_to_group(self):
+        self.form.signup.user_group_template = Expression('string:staff')
+        signup = self.form.signup
+        fields = self.form._getFieldObjects()
+        request = self.LoadRequestForm(fullname='test name', username='test1', email='test@mail.com', )
+        signup.onSuccess(fields, request)
+        groups = api.group.get_groups(username='test1')
+        self.assertTrue('staff' in [group.id for group in groups],'User not created with staff group')
+
+    def test_add_to_dynamic_group(self):
+        self.form.signup.user_group_template = Expression('string:${department}')
+        signup = self.form.signup
+        fields = self.form._getFieldObjects()
+        request = self.LoadRequestForm(fullname='test name', username='test1', email='test@mail.com',department='IT' )
+        signup.onSuccess(fields, request)
+        groups = api.group.get_groups(username='test1')
+        self.assertTrue('IT' in [group.id for group in groups],'User not created with IT group')
+
+    def test_invalid_domain_name_validation(self):
+        self.form.signup.user_group_template = Expression('string:staff')
+        self.form.signup.email_domain_verification = True
+        signup = self.form.signup
```

Comment:
even though you preset the data I think to add an assert here to show the preset data so you can read the test and understand what the preset data is. ie assert the group domain = mygroup@mail.com